test/functional/lua/buffer_updates_spec.lua

```diff
@@ -1250,6 +1250,224 @@ describe('lua: nvim_buf_attach on_bytes', function()
       }
     end)
 
+    it('on_bytes sees modified buffer during substitute', function()
```

Comment:
?? Why should this be supported? We should not be emitting buffer updates during the "live preview". Only after the substitute is confirmed. The "live preview" is NOT an edit of any kind, it is just a preview.

If a plugin wants to do stuff during the preview it should implement an 'inccommand' preview handler (`:help :command-preview`).

I don't see anything like that in these tests so I assume that is not the case here.